### PR TITLE
Terraform: Create & configure lambda function and deploy

### LINKED ISSUE #193: Musfik - Add Lambda function with required resources & API gateway
Create Lambda function with terraform & add zip archive file for lambda

### PR DESCRIPTION
**Pull request by musfiksifat**
Fixes #193

- [ ] Does not depend on other PRs
- [x] Depends on other PR being merged first: #169 

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
Create lambda resource using terraform
## What was done
- Created a new lambda resource
- Added lambda execution role
- Moved all serverless part in a separate directory to create and add zip archive for lambda
- Added terraform directory to .gitignore

## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: terraform-aws/terraform.tfstate [hunk 1]
```diff
@@ -0,0 +1,146 @@
+{
```
[the remark this answers is not in the document]
Comment (musfiksifat) [reply terraform-aws/terraform.tfstate]: resolved by [de38387](https://github.com/CraftsmenLtd/coe-batch-q322/pull/203/commits/de3838765253aff12181b7da9e029eb28afbfec6)

File: terraform-aws/terraform.tfstate.backup [hunk 2]
```diff
@@ -0,0 +1,146 @@
+{
```
[the remark this answers is not in the document]
Comment (musfiksifat) [reply terraform-aws/terraform.tfstate.backup]: resolved by [5716790](https://github.com/CraftsmenLtd/coe-batch-q322/pull/203/commits/57167907ba9d358aeda616e5989a9c8e7abc76a0)

File: terraform-aws/variables.tf [hunk 3]
```diff
@@ -0,0 +1,5 @@
+variable "aws_region" {
+  description = "AWS region"
+  type        = string
+  default     = "ap-northeast-1"
```
[the remark this answers is not in the document]
Comment (musfiksifat) [reply terraform-aws/variables.tf:4]: resolved by [5716790](https://github.com/CraftsmenLtd/coe-batch-q322/pull/203/commits/57167907ba9d358aeda616e5989a9c8e7abc76a0)

File: terraform-aws/lambda.tf [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+  runtime = "nodejs14.x"
+
+  environment {
+    variables = {
+      S3_BUCKET_NAME     = "musfik-serverless-bucket2"
+      DYNAMODB_TABLENAME = "Musfik-serverless-table "
+      SNS_TOPIC_NAME     = "Musfik-sns-topic"
+      SNS_TOPIC_ARN      = "arn:aws:sns:ap-northeast-1:385187169402:Musfik-sns-topic"
+      WEBHOOK_URL        = "https://discord.com/api/webhooks/1004259829667336223/pOtmlTZikCbs8OG5IV-2fTOayuNPggM2ZZNukcM7jIJ-PONqIDHV1ous4VU0fBucGS2Y"
```
[the remark this answers is not in the document]
Comment (musfiksifat) [reply terraform-aws/lambda.tf]: resolved by [5716790](https://github.com/CraftsmenLtd/coe-batch-q322/pull/203/commits/57167907ba9d358aeda616e5989a9c8e7abc76a0)

**Review by mhazad — APPROVED** [review]
LGTM

**Pull request merged**